controllers/kubeadmconfig_controller.go

```diff
@@ -398,8 +398,8 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *clusterv1.Cluster,
 	}
 
 	// if BootstrapToken already contains an APIServerEndpoint, respect it; otherwise inject the APIServerEndpoint endpoint defined in cluster status
-	//TODO(fp) might be we want to validate user provided APIServerEndpoint and warn/error if it doesn't match the api endpoint defined at cluster level
 	apiServerEndpoint := config.Spec.JoinConfiguration.Discovery.BootstrapToken.APIServerEndpoint
+	needValidateAPIEndpoints := true
```

Comment:
We could do away with this boolean and always validate, right? I think the optimization of running this validation code is not worth introducing the boolean and we should validate every time regardless.